staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -290,6 +302,17 @@ func (r *crdHandler) updateCustomResourceDefinition(oldObj, newObj interface{})
 	r.customStorageLock.Lock()
 	defer r.customStorageLock.Unlock()
 
+	// Add CRD to the establishing controller queue.
+	// For HA clusters, we want to prevent race conditions when changing status to Established,
+	// so we want to be sure that CRD is Installing at least for 5 seconds before Establishing it.
+	if apiextensions.IsCRDConditionTrue(newCRD, apiextensions.Installing) {
```

Comment:
`// TODO: find a real HA safe checkpointing mechanism instead of an arbitrary wait`